File: PATMAP/PATMAP/PATMAP/Reports.ascx.vb
Imports System.Data
Imports System.Configuration
Imports System.Collections
Imports System.Web
Imports System.Web.Security
Imports System.Web.UI
Imports System.Web.UI.WebControls
Imports System.Web.UI.WebControls.WebParts
Imports System.Web.UI.HtmlControls
Imports System.Collections.Specialized
Imports System.Text

Partial Public Class Reports
    Inherits System.Web.UI.UserControl

    Protected Sub Page_Load(ByVal sender As Object, ByVal e As EventArgs)
        Dim getMapValues As String = (("getMapValues('" + ConfigurationManager.AppSettings("MunicipalitiesLayerName") & "','") + ConfigurationManager.AppSettings("SchoolDivisionsLayerName") & "','") + ConfigurationManager.AppSettings("ParcelLayerName") & "');"
        Me.btnOpenTables.OnClientClick = getMapValues
        Me.btnOpenGraphs.OnClientClick = getMapValues
    End Sub

    Private Sub setSession()
        Dim formVars As NameValueCollection = Me.Page.Request.Form

        If formVars("MunicipalityID") IsNot Nothing AndAlso formVars("MunicipalityID").Length > 0 Then
            Session.Add("Municipalities", formVars("MunicipalityID"))
        End If

        If formVars("SchoolID") IsNot Nothing AndAlso formVars("SchoolID").Length > 0 Then
            Session.Add("SchoolDistricts", formVars("SchoolID"))
        End If

        If formVars("ParcelID") IsNot Nothing AndAlso formVars("ParcelID").Length > 0 Then
            Session.Add("ParcelID", formVars("ParcelID"))
        End If

        ''If MapSettings.MapPropertyClassFilters IsNot Nothing AndAlso MapSettings.MapPropertyClassFilters.Count = 1 Then
        ''    Session.Add("TaxClass", MapSettings.MapPropertyClassFilters(0))
        ''End If

        ''Session.Add("TaxType", If(MapSettings.MapAnalysisLayer.Equals("Municipalities"), "1", "2"))

        ''If MapSettings.TaxStatusFilters IsNot Nothing AndAlso MapSettings.TaxStatusFilters.Count = 1 Then
        ''    Dim taxStatusFilter As New StringBuilder()
        ''    For Each taxStatus As String In MapSettings.TaxStatusFilters
        ''        taxStatusFilter.Append(taxStatus)
        ''        taxStatusFilter.Append(",")
        ''    Next
        ''    Session.Add("TaxStatusFilters", taxStatusFilter.ToString())
        ''End If
    End Sub

    Protected Sub btnOpenTables_Click(ByVal sender As Object, ByVal e As EventArgs)
        setSession()
        ''If BoundaryChangeSettings.BoundaryChangeState = BoundaryChangeSettings.BOUNDARY_CHANGE_STATE.NONE Then
        ''    Me.Page.ClientScript.RegisterStartupScript(Me.[GetType](), "OpenReport", "parent.document.location='assmnt/tables.aspx'", True)
        ''ElseIf BoundaryChangeSettings.BoundaryChangeState = BoundaryChangeSettings.BOUNDARY_CHANGE_STATE.LTT Then
        ''    Me.Page.ClientScript.RegisterStartupScript(Me.[GetType](), "OpenReport", "parent.document.location='taxtools/tables.aspx'", True)
        ''Else
        ''    Me.Page.ClientScript.RegisterStartupScript(Me.[GetType](), "OpenReport", "parent.document.location='boundary/tables.aspx'", True)
        ''End If
    End Sub

    Protected Sub btnOpenGraphs_Click(ByVal sender As Object, ByVal e As EventArgs)
        setSession()
        ''If BoundaryChangeSettings.BoundaryChangeState = BoundaryChangeSettings.BOUNDARY_CHANGE_STATE.NONE Then
        ''    Me.Page.ClientScript.RegisterStartupScript(Me.[GetType](), "OpenReport", "parent.document.location='assmnt/graphs.aspx'", True)
        ''ElseIf BoundaryChangeSettings.BoundaryChangeState = BoundaryChangeSettings.BOUNDARY_CHANGE_STATE.LTT Then
        ''    Me.Page.ClientScript.RegisterStartupScript(Me.[GetType](), "OpenReport", "parent.document.location='taxtools/graphs.aspx'", True)
        ''Else
        ''    Me.Page.ClientScript.RegisterStartupScript(Me.[GetType](), "OpenReport", "parent.document.location='boundary/graphs.aspx'", True)
        ''End If
    End Sub

End Class

File: PATMAPGIS_2012/PATMAPGIS_2012/Reports.ascx.cs
using System;
using System.Data;
using System.Configuration;
using System.Collections;
using System.Web;
using System.Web.Security;
using System.Web.UI;
using System.Web.UI.WebControls;
using System.Web.UI.WebControls.WebParts;
using System.Web.UI.HtmlControls;
using System.Collections.Specialized;
using System.Text;

public partial class Reports : System.Web.UI.UserControl
{
    protected void Page_Load(object sender, EventArgs e)
    {
        string getMapValues = "getMapValues('" + ConfigurationManager.AppSettings["MunicipalitiesLayerName"] + "','" + ConfigurationManager.AppSettings["SchoolDivisionsLayerName"] + "','" + ConfigurationManager.AppSettings["ParcelLayerName"] + "');";
        this.btnOpenTables.OnClientClick = getMapValues;
				this.btnOpenGraphs.OnClientClick = getMapValues;
				if (System.Web.HttpContext.Current.Session["BoundaryChangeStale"] != null)
				{
					this.btnOpenGraphs.Visible = false;
				} else {
					this.btnOpenGraphs.Visible = true;
				}
    }

    private void setSession()
    {
        NameValueCollection formVars = this.Page.Request.Form;

        if (formVars["MunicipalityID"] != null && formVars["MunicipalityID"].Length > 0)
        {
            Session.Add("Municipalities", formVars["MunicipalityID"]);
        }

        if (formVars["SchoolID"] != null && formVars["SchoolID"].Length > 0)
        {
            Session.Add("SchoolDistricts", formVars["SchoolID"]);
        }

        if(formVars["ParcelID"] != null && formVars["ParcelID"].Length > 0)
        {
             Session.Add("ParcelID", formVars["ParcelID"]);
        }

        if (MapSettings.MapPropertyClassFilters != null && MapSettings.MapPropertyClassFilters.Count == 1)
        {
            Session.Add("TaxClass", MapSettings.MapPropertyClassFilters[0]);
        }

        Session.Add("TaxType", MapSettings.MapAnalysisLayer.Equals("Municipalities") ? "1" : "2");

        if (MapSettings.TaxStatusFilters != null && MapSettings.TaxStatusFilters.Count == 1)
        {
            StringBuilder taxStatusFilter = new StringBuilder();
            foreach (string taxStatus in MapSettings.TaxStatusFilters)
            {
                taxStatusFilter.Append(taxStatus);
                taxStatusFilter.Append(",");
            }
            Session.Add("TaxStatusFilters", taxStatusFilter.ToString());
        }
    }

    protected void btnOpenTables_Click(object sender, EventArgs e)
    {
        setSession();
        if (BoundaryChangeSettings.BoundaryChangeState == BoundaryChangeSettings.BOUNDARY_CHANGE_STATE.NONE)
        {
            //this.Page.ClientScript.RegisterStartupScript(this.GetType(), "OpenReport", "parent.parent.document.location='assmnt/tables.aspx'", true);
					this.Page.ClientScript.RegisterStartupScript(this.GetType(), "OpenReport", "openTablesLink('assmnt');", true);
        }
        else if (BoundaryChangeSettings.BoundaryChangeState == BoundaryChangeSettings.BOUNDARY_CHANGE_STATE.LTT)
        {
            //this.Page.ClientScript.RegisterStartupScript(this.GetType(), "OpenReport", "parent.document.location='taxtools/tables.aspx'", true);
					this.Page.ClientScript.RegisterStartupScript(this.GetType(), "OpenReport", "openTablesLink('taxtools');", true);
        }
        else
        {
           //this.Page.ClientScript.RegisterStartupScript(this.GetType(), "OpenReport", "parent.document.location='boundary/tables.aspx'", true);
					this.Page.ClientScript.RegisterStartupScript(this.GetType(), "OpenReport", "openTablesLink('boundary');", true);
        }
    }

    protected void btnOpenGraphs_Click(object sender, EventArgs e)
    {
        setSession();
        if (BoundaryChangeSettings.BoundaryChangeState == BoundaryChangeSettings.BOUNDARY_CHANGE_STATE.NONE)
        {
            //this.Page.ClientScript.RegisterStartupScript(this.GetType(), "OpenReport", "parent.document.location='assmnt/graphs.aspx'", true);
					this.Page.ClientScript.RegisterStartupScript(this.GetType(), "OpenReport", "openGraphsLink('assmnt');", true);
        }
        else if (BoundaryChangeSettings.BoundaryChangeState == BoundaryChangeSettings.BOUNDARY_CHANGE_STATE.LTT)
        {
            //this.Page.ClientScript.RegisterStartupScript(this.GetType(), "OpenReport", "parent.document.location='taxtools/graphs.aspx'", true);
					this.Page.ClientScript.RegisterStartupScript(this.GetType(), "OpenReport", "openGraphsLink('taxtools');", true);
        }
        else
        {
            //this.Page.ClientScript.RegisterStartupScript(this.GetType(), "OpenReport", "parent.document.location='boundary/graphs.aspx'", true);
					this.Page.ClientScript.RegisterStartupScript(this.GetType(), "OpenReport", "openGraphsLink('boundary');", true);
        }
    }

}
